src/Foundation/NSUrlSessionHandler.cs

```diff
@@ -1083,6 +1089,18 @@ void DidReceiveChallengeImpl (NSUrlSession session, NSUrlSessionTask task, NSUrl
 					}
 					return;
 				}
+                if (sessionHandler.ClientCertificateOptions == ClientCertificateOption.Manual && challenge.ProtectionSpace.AuthenticationMethod == NSUrlProtectionSpace.AuthenticationMethodClientCertificate)
+                {
+					var certificate = System.Net.Security.CertificateHelper.GetEligibleClientCertificate(sessionHandler.ClientCertificates);
+					if (certificate != null)
+					{
+						var cert = new SecCertificate(certificate);
+						var identity = SecIdentity.Import(certificate);
+						var credential = new NSUrlCredential(identity, new SecCertificate[] { cert }, NSUrlCredentialPersistence.ForSession);
```

Comment:
Not familiar with this API, but can NSUrlCredential accept the entire collection here (with/without filtering)?

Reply:
No you can only encrypt with one certificate. Encrypting with multiple makes no sense.  Also matches how other handlers do it

Reply:
The usage here doesn't quite align with https://developer.apple.com/documentation/foundation/nsurlcredential/1428192-credentialwithidentity. What if the user provided intermediate certificates that need to be passed along?

Reply:
When the user picks a certificate for authentication there’s just that one. I’ve never seen a chain be needed here and other .net handlers doesn’t do this either. This PR brings equivalence with other .net handlers (but with the benefit of using the iOS session which support caching and much more)

Reply:
Without providing the intermediate certificates, the server is required to have all the intermediate certificates already installed. This may be the common case, but wanted to point out this is not required and this code will fail in that scenario.